### PR TITLE
Fix RATIFY conformance

### PR DESCRIPTION
**Pull request by aniketd**
# Description

This still requires the corresponding `HSSet` de-duplication to work.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

**Comment by MaximilianAlgehed** [comment]
What is the reason the tests are failing now?

**Comment by aniketd** [comment]
They seem to be passing. Perhaps only a review is required? @MaximilianAlgehed 

cc. @Soupstraw

**Comment by MaximilianAlgehed** [comment]
I mean before this change - what's the underlying problem you're fixing here (I'm curious and trying to understand)

File: cabal.project [hunk 1]
```diff
@@ -13,8 +13,7 @@ repository cardano-haskell-packages
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-executable-spec.git
-  tag: 81b976c40c2726998a4828220231acf75b3a685e
-  --sha256: sha256-1CiTiLzROoHz8GxVPCouofUFzwzWUclQUq5AthtwPz4=
```
Comment (lehins) [inline cabal.project:17]: You should update the sha256 instead of removing it

**Review by lehins — COMMENTED** [review]
Looks good.
Let's wait for @Soupstraw to approve it before merging.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway.hs [hunk 2]
```diff
@@ -478,10 +473,13 @@ ratifyStateSpec ::
...
   Specification fn (RatifyState Conway)
-ratifyStateSpec ConwayRatifyExecContext {crecGovActionMap} =
+ratifyStateSpec _ =
   constrained $ \st ->
-    match st $ \_ _ expired _ ->
-      expired `subset_` dom_ (lit crecGovActionMap)
+    match st $ \_ enacted expired _ ->
+      mconcat
+        [ assert $ enacted ==. lit mempty
+        , assert $ expired ==. lit mempty
```
Comment (Soupstraw) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway.hs:481]: These seem to be overconstraining the state. We want to get cases where there are values in `enacted` and `expired`, but right now it's only generating cases where these fields are empty.

**Review by Soupstraw — CHANGES_REQUESTED** [review]
Looks good overall, but I think the generators are overconstrained.

**Comment by Soupstraw** [comment]
@MaximilianAlgehed, the confomance tests for RATIFY didn't pass before, but this PR aims to fix this problem.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway.hs (hunk 2 above)
Comment (aniketd) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway.hs:481]: We discussed this on a call when you were on holiday 😜. Let's discuss in a huddle today?

**Review by Soupstraw — APPROVED** [review]

**Pull request merged**